### PR TITLE
Change into a plain-python package, using tox for tests, with Action CI

### LINKED ISSUE #109: Don't depend on ament Python linters via a ROS2 installation
## Description
In order to remove this tool from the ROS2 build ecosystem, we can't depend on the ament linters via a ROS2 installation. Instead of relying on them that way, install via pip VCS option

## Related Issues
Part of Epic https://github.com/ros-tooling/cross_compile/issues/102

## Completion Criteria
cross_compile does not depend on any `ament` packages being installed via `apt`

## Implementation Notes/Suggestions
https://discourse.ros.org/t/ament-lint-for-pure-python-packages/12247/2 based on this conversation we should be able to just install the ament_lint packages with pip via its vcs options, e.g.

```
pip3 install -e "git+https://github.com/ament/ament_lint.git@0.8.1#egg=ament_lint&subdirectory=ament_lint" 
```

### LINKED ISSUE #110: Integrate GitHub Action for pure python package to build and test on OS matrix
## Description
Instead of using our custom ROS2 CI GH Actions, set up this repo to use a CI Action for regular python packages.

## Related Issues
Part of Epic https://github.com/ros-tooling/cross_compile/issues/102
Depends on https://github.com/ros-tooling/cross_compile/issues/109

## Completion Criteria
* PRs are built with python action
* PRs are tested with python action
* Artifact of the build can be used in local virtualenv

## Implementation Notes/Suggestions
https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

### LINKED ISSUE #111: Remove all ROS-specific infrastructure
## Description
Make this package a pure-python package.

After doing this, the package will no longer be testable via `colcon`. Since all tests were being invoked via colcon, successful completion of this story means introducing a new way to call tests. We should use [Tox](https://tox.readthedocs.io/en/latest/) for this.

## Related Issues
Part of Epic https://github.com/ros-tooling/cross_compile/issues/102
Depends on https://github.com/ros-tooling/cross_compile/issues/109

## Completion Criteria
* No ros2-specific infrastructure exists anymore
* All existing tests can be run successfully via Tox for the python versions we claim to support

Things that will need to be removed or changed:
*  `resource/`
*  `cross_compile.repos`
*  `requirements.txt`
*  `setup.py`
* `setup.cfg`
* `package.xml`
* `README.md`

### PR DESCRIPTION
**Pull request by emersonknapp**
CI:
* Changing the workflows for this package to use basic python tools and run tox tests.
* Remove the linter check because the test suite now runs the linters (on every target platform, getting a broader range of testing)

Package:
* This PR makes cross_compile into a plain-python package, removing the ROS infrastructure around it.
* It uses tox to run tests in place of `colcon test`, depending on the ament linters via a pip VCS URL

Fixes #109
Fixes #110 
Fixes #111 

Notes:
* The total codecov goes down a bit with this check, but it's a side effect of reorganizing
* After landing, I will change the branch protections to change the required checks to the ones being introduced here, so "Required" checks below should be ignored.

### CODE REVIEW

File: .github/workflows/e2e_test.yml [hunk 1]
```diff
@@ -14,21 +14,24 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        supported_platforms: [aarch64, armhf]
-        supported_os: [ubuntu, debian]
-        rosdistro: [dashing, eloquent, melodic]
+        target_arch: [aarch64, armhf]
+        target_os: [ubuntu , debian]
```
Comment (zmichaels11) [inline .github/workflows/e2e_test.yml]: space?

File: test/test_entrypoint.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 """Unit tests for the `create_cc_sysroot.py` script."""
 
-from cross_compile.ros2_cross_compile import parse_args
+from cross_compile.ros_cross_compile import parse_args
```
Comment (piraka9011) [inline test/test_entrypoint.py]: Aren't we removing all ros-deps?

Comment (emersonknapp) [reply test/test_entrypoint.py]: This is importing a module from this package - i renamed the source file which is why this line is changing

File: .github/workflows/e2e_test.yml (hunk 1 above)
Comment (emersonknapp) [reply .github/workflows/e2e_test.yml]: good catch, thanks

Comment (thomas-moulard) [reply .github/workflows/e2e_test.yml]: +1

**Comment by thomas-moulard** [comment]
Could you also update ros-security/aws-oncall accordingly? (if needed)

**Pull request merged**